atlassian_jwt_auth/contrib/django/decorators.py

```diff
@@ -4,10 +4,54 @@
 from django.http.response import HttpResponse
 
 import atlassian_jwt_auth
-from .utils import parse_jwt, verify_issuers
+from .utils import parse_jwt, verify_issuers, _build_response
 from ..server.helpers import _requires_asap
 
 
+def validate_asap(issuers=None, subjects=None, required=True):
+    """Decorator to allow endpoint-specific ASAP authorization, assuming ASAP
+    authentication has already occurred.
+
+    :param list issuers: A list of issuers that are allowed to use the
+        endpoint.
+    :param subject: A list of subjects or a function to determine allowed
+        subjects for the endpoint.
+    :param boolean required: Whether or not to require ASAP on this endpoint.
+        Note that requirements will be still be verified if claims are present.
+    """
+    def validate_asap_decorator(func):
+        @wraps(func)
+        def validate_asap_wrapper(request, *args, **kwargs):
+            asap_claims = getattr(request, 'asap_claims', {})
+            if required and not asap_claims:
+                message = 'Unauthorized: Invalid or missing token'
+                response = HttpResponse(message, status=401)
+                response['WWW-Authenticate'] = 'Bearer'
+                return response
+
```

Comment:
I suggest adding a check here for the case of `not required and not asap_claims`.

Reply:
I addressed this in fea56c1829e83497fea0b74d620116e89da545c7; are there any other issues preventing this from being merged?

Reply:
Yes some pep8 fixes and the addition of tests to check the new code.